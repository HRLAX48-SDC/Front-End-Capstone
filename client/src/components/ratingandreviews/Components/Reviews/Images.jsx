import React, { useState, useEffect} from 'react';
import { Card, Button, Modal, Image, Row, ModalBody } from 'react-bootstrap'



function Images ({ image }) {
    const[imageLarge, setImageLarge]= useState(false)

    return (
        <>
        <Image className="review-photos" thumbnail src={image.url} onClick={() => setImageLarge(true) }/>
        <Modal show={imageLarge} centered size="large" onHide={() => setImageLarge(false)}>
            <ModalBody>
                <Image src={image.url} thumbnail/>
            </ModalBody>
        </Modal>
        </>
    )
}

export default Images;
