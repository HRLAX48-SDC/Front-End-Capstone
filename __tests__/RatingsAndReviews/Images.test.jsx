import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Images from '../../client/src/components/ratingandreviews/Components/Reviews/Images.jsx'

describe('<Images />', () => {
    const imageTest = {id: 2140839, url: "https://i.imgur.com/760q3rR.jpg"};
    it('should correctly render a single image thumbnail', () =>{
        const image = renderer
        .create(<Images image={imageTest.url}/>)
        .toJSON();
        expect(image).toMatchSnapshot()

    });
});