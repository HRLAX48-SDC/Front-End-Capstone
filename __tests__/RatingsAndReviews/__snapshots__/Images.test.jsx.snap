// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Images /> should correctly render a single image thumbnail 1`] = `
<img
  className="review-photos img-thumbnail"
  onClick={[Function]}
/>
`;
